tests/api/serve.js

```diff
@@ -55,9 +55,14 @@ const processDates = (obj, usingHourly = false) => {
         // If we are parsing hourly forcast data, and the key is either the
         // start or end time, then align the output to the start of the given
         // hour.
+        //
+        // Alternatively, if we are not parsing hourly data but our parent key
+        // is "values" and our key is "validTime", then this is the hourly data
+        // stuffed inside the gridpoints API return. Do the same thing.
         if (
-          usingHourly &&
-          ["startTime", "endTime", "validTime"].includes(key)
+          (usingHourly &&
+            ["startTime", "endTime", "validTime"].includes(key)) ||
+          (!usingHourly && parent === "values" && key === "validTime")
```

Comment:
gosh, maybe it's just Monday but I'm having a hard time parsing this. can we pull out these as descriptive boolean variables? Something like:

```js
  const needsAlignment = ["startTime", "endTime", "validTime"].includes(key);
  const isHourlyInGridPoints = !usingHourly && parent === "values" && key === "validTime";

  if (usingHourly && needsAlignment || isHourlyInGridPoints) ...
```

and if so, is that boolean logic correct?

Reply:
Oof, yeah, this is gnarly. And I wrote it long enough ago that I'm not certain anymore... 😅 

I'll spend some time walking through it today to understand it again and then I'll clean it up so it's not as hairy. Good catch!

Reply:
This was way too convoluted. I did understand it after I sat with it for a minute, but it was a hack because I didn't think very long about how to determine if the data being processed was hourly forecast data. It turns out, when I spent a little time on that, there was a pretty easy regex I could use down below and then all of this wonkadoo `if` logic just goes away.